control-plane/api-gateway/gatekeeper/gatekeeper.go

```diff
@@ -53,20 +57,27 @@ func (g *Gatekeeper) Upsert(ctx context.Context, gateway gwv1beta1.Gateway, gcc
 }
 
 // Delete removes the resources for handling routing of network traffic.
-func (g *Gatekeeper) Delete(ctx context.Context, nsname types.NamespacedName) error {
-	if err := g.deleteRole(ctx, nsname); err != nil {
+// This is done in the reverse order of Upsert due to dependencies between resources.
```

Comment:
Calling this out as the reason for re-organizing the order of delete logic below